modules/e2ee/OlmAdapter.js

```diff
@@ -19,17 +21,30 @@ const OLM_MESSAGE_TYPES = {
     KEY_INFO: 'key-info',
     KEY_INFO_ACK: 'key-info-ack',
     SESSION_ACK: 'session-ack',
-    SESSION_INIT: 'session-init'
+    SESSION_INIT: 'session-init',
+    SAS_START: 'sas-start',
+    SAS_ACCEPT: 'sas-accept',
+    SAS_KEY: 'sas-key',
+    SAS_MAC: 'sas-mac'
 };
 
+const OLM_SAS_NUM_BYTES = 6;
+const OLM_KEY_VERIFICATION_MAC_INFO = 'Jitsi-KEY_VERIFICATION_MAC';
+const OLM_KEY_VERIFICATION_MAC_KEY_IDS = 'Jitsi-KEY_IDS';
+
 const kOlmData = Symbol('OlmData');
 
 const OlmAdapterEvents = {
-    OLM_ID_KEY_READY: 'olm.id_key_ready',
     PARTICIPANT_E2EE_CHANNEL_READY: 'olm.participant_e2ee_channel_ready',
-    PARTICIPANT_KEY_UPDATED: 'olm.partitipant_key_updated'
+    PARTICIPANT_SAS_KEY_AVAILABLE: 'olm.participant_sas_key_available',
+    PARTICIPANT_SAS_READY: 'olm.participant_sas_ready',
+    PARTICIPANT_KEY_UPDATED: 'olm.partitipant_key_updated',
+    PARTICIPANT_VERIFICATION_COMPLETED: 'olm.participant_verification_completed'
 };
 
+const keysMacMismatchedError = 'keys MAC mismatch';
```

Comment:
these need to be exporsded somewhere for the app to use and be something like "e2ee.sas.keys-mac-mismatch" so it can be translated easily